pkg/log/log.go

```diff
@@ -117,7 +120,24 @@ func (l *logger) Print(msg string, fields ...field.M) {
 
 	entry := log.WithFields(logFields)
 	if l.err != nil {
-		entry = entry.WithError(l.err)
+		switch e := l.err.(type) {
```

Comment:
It appears that the effect is to change AWS errors into fields and never set the error in the entry for those.  Is that the intent?

It would be more efficient and I think easier to read if adding the AWS fields was done slightly above here as part of filling in logFields before line 121.  A check of `if e, ok := l.err.(awserr.Error); ok {` checks for nil and the type match (I think).

Reply:
To answer your first question: here in case of awserr, we are doing the work for entry.WithError(). We are extracting the well know fields and assigning them to the log to make it more descriptive and clear. Otherwise, entry.WithError(err) internally calls entry.WithFields(ErrorKey, err).


About moving the aws fields before line 121, i really liked the suggestion, let me test with nil value and push something.